Library/Homebrew/cleanup.rb

```diff
@@ -348,32 +348,34 @@ def cleanup_lockfiles(*lockfiles)
     end
 
     def cleanup_portable_ruby
-      system_ruby_version =
-        Utils.popen_read("/usr/bin/ruby", "-e", "puts RUBY_VERSION")
-             .chomp
-      use_system_ruby = (
-        Gem::Version.new(system_ruby_version) >= Gem::Version.new(RUBY_VERSION)
-      ) && !Homebrew::EnvConfig.force_vendor_ruby?
-      vendor_path = HOMEBREW_LIBRARY/"Homebrew/vendor"
-      portable_ruby_version_file = vendor_path/"portable-ruby-version"
-      portable_ruby_version = if portable_ruby_version_file.exist?
-        portable_ruby_version_file.read
-                                  .chomp
+      rubies = [which("ruby"), which("ruby", ENV["HOMEBREW_PATH"])]
+      system_ruby = Pathname.new("/usr/bin/ruby")
+      rubies << system_ruby if system_ruby.exist?
+
+      use_system_ruby = if Homebrew::EnvConfig.force_vendor_ruby?
+        false
+      else
+        check_ruby_version = HOMEBREW_LIBRARY_PATH/"check_ruby_version.rb"
+        rubies.uniq.any? do |ruby|
+          system ruby, "--enable-frozen-string-literal", "--disable=gems,did_you_mean,rubyopt",
+                 check_ruby_version, HOMEBREW_REQUIRED_RUBY_VERSION
```

Comment:
Hmm, not sure why `HOMEBREW_REQUIRED_RUBY_VERSION` seems to be unset in the Docker container 🤔

Reply:
I debugged this locally -- it's a `nil` in `rubies` that causes that message, not `HOMEBREW_REQUIRED_RUBY_VERSION`. I added `.compact` to fix this